### PR TITLE
fix: Doc 29 ground-roll geometry, landing rollout and Annex 16 bandsharing (phase 2, wave E)

### PR DESCRIPTION
**Pull request by jmrplens**
## Summary

First quality wave of the overhaul phase 2: a coefficient-by-coefficient review of the aircraft domain against ECAC Doc 29 Vol 2, ICAO Annex 16 App. 2 and SAE ARP 5534 confirmed **every transcribed constant correct** and found three HIGH defects plus refinements, all validated against the Doc 29 Vol 3 Part 1 reference workbook.

### Fixed (HIGH)
- **Behind takeoff ground roll**: lateral attenuation and installation effect now use the §4.5.5 nearest-end geometry (`β₁ = asin(z₁/d₁)`, `ℓ = OC₁`); the old track-perpendicular geometry zeroed both on the extended centreline (~**10 dB SEL overprediction**, workbook JETFDC/R03).
- **Eq. 4-13b**: runway segments use the average speed `(V₁+V₂)/2`; `V = 0` start-of-roll waypoints accepted.
- **Landing rollout modelled**: new `landing_roll` mask; ahead of it: `ds` baseline, reduced fraction Eq. 4-21b, nearest-end geometry, no directivity (workbook JETFAS/R05).

### Fixed (MEDIUM) / Added (LOW)
- Equivalent-level-path elevation angle on inclined segments (Fig. 4-6); nearest-end Lmax geometry behind/ahead; 30 m NPD floor (§4.2, also fixes a crash for receivers under the path).
- **Annex 16 bandsharing ΔB** (App. 2 §4.4.2/4.4.3 per the ETM guidance): applied before the 10 dB-down window, included in PNLTM and EPNL, exposed as `EPNLResult.bandsharing_adjustment`. No worked ΔB example exists in the ETM (checked); hand-built oracle, gap registered.
- Per-segment `bank` parameter (§4.5.2 sign convention, port/starboard verified antisymmetric); `tone_correction(start_band=0)` for the helicopter 50 Hz procedure; docstring notes for the power-extrapolation licence, the exact EPNL constant and the degenerate-path `-inf`.

## Validation

- **Seven branch-covering receptor events** of the reference workbook hardcoded as a per-segment assembly oracle (29-33 segments each) plus B-1 event totals (energy sum exact to 3 thousandths of a dB); new conformance check, **195/195**.
- Golden baselines regenerated for the two airport cases: intended numeric change, max drift **0.004 dB** (event) / **0.020 dB** (contour) from the equivalent-path angle; contour figure variants regenerated.
- Full suite **2662 tests green**; ruff + mypy clean.

<a href="https://cubic.dev/pr/jmrplens/phonometry/pull/164?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **New Features**
  - Added landing-roll airport-noise modeling, including reduced-noise calculations, nearest-end geometry selection, and per-segment bank-angle handling.
  - Enhanced tone correction for helicopter 50 Hz procedures and added ICAO EPNL bandsharing adjustment reporting.
- **Bug Fixes**
  - Corrected behind/nearby ground-roll geometry, runway speed handling, and exposure/magnitude computations; added a 30 m minimum for NPD lookups.
  - Improved speech intelligibility bandwidth-importance normalization and resolved edge-case receptor calculations.
- **Tests**
  - Expanded conformance and golden regression coverage with new reference scenario checks.
- **Documentation**
  - Updated aircraft-noise guides and regenerated conformance results (now 195/195 checks passing).

### CODE REVIEW

File: src/phonometry/aircraft/aircraft_noise.py [hunk 1]
```diff
@@ -318,10 +335,18 @@ def epnl_from_pnlt(
...
 
-    pnltm = float(np.max(p))
+    km = int(np.argmax(p))
+    delta_b = 0.0
+    if tone_corrections is not None:
+        c = np.atleast_1d(np.asarray(tone_corrections, dtype=np.float64))
+        if c.shape != p.shape or not np.all(np.isfinite(c)):
+            raise ValueError("'tone_corrections' must match 'pnlt' in length and be finite.")
+        window = c[max(0, km - 2):km + 3]
+        delta_b = max(0.0, float(np.mean(window)) - float(c[km]))
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/aircraft/aircraft_noise.py]: Good catch. The window is now time based: record midpoints are computed from dt (scalar or per record) and the average takes every record whose midpoint lies within 1 s of the PNLTM record, which reproduces the five-record slice for the uniform 0.5 s cadence and adapts to any other spacing. Covered by test_bandsharing_window_scales_with_record_duration.

File: src/phonometry/aircraft/airport_noise.py [hunk 2]
```diff
@@ -469,45 +485,89 @@ def _event_level_core(
...
+        # analogues ahead). The engine-installation depression angle follows the
+        # same equivalent geometry on those branches.
+        d1 = float(np.linalg.norm(obs - s1))
+        d2 = float(np.linalg.norm(obs - s2))
+        z1 = float(s1[2] - obs[2])
+        z2 = float(s2[2] - obs[2])
+        beta1 = float(np.degrees(np.arcsin(np.clip(z1 / d1, -1.0, 1.0)))) if d1 > 0 else 90.0
+        beta2 = float(np.degrees(np.arcsin(np.clip(z2 / d2, -1.0, 1.0)))) if d2 > 0 else 90.0
+        oc1 = float(np.sqrt(max(d1**2 - z1**2, 0.0)))
+        oc2 = float(np.sqrt(max(d2**2 - z2**2, 0.0)))
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/aircraft/airport_noise.py]: Applied: the nearest-end horizontal distance is now np.hypot of the ground-plane offsets, which is the same quantity without the cancellation-prone subtraction.

**Comment by jmrplens** [comment]
Review follow-ups applied in the last three commits:

- SonarCloud: the segment loop's cognitive complexity was reduced by extracting the nearest-end attenuation geometry (4.5.5) and the noise-fraction selection (Eq 4-20/4-21a/4-21b) into dedicated helpers. The 14-parameter S107 warnings on event_level and noise_contour are registered as a documented exception in sonar-project.properties: the signatures mirror the Doc 29 single-event parameter set and every extra is a keyword-only optional (the duplicate multicriteria key this introduced was merged into a single list, restoring the pre-existing facade exception).
- gemini: bandsharing window is now time based (records within 1 s of PNLTM) instead of a fixed five-record slice, and the nearest-end horizontal distance uses np.hypot.
- CodeRabbit (outside diff): noise_contour now accepts bank with the same validation as event_level and threads it to the grid evaluation.

Full suite 2664 passed, conformance 195/195.
